components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1383,6 +1337,17 @@ where
 
         self.fsm.peer.force_leader = Some(ForceLeaderState::ForceLeader { failed_stores });
         self.fsm.has_ready = true;
+        let mut leaders = forced_leaders.lock().unwrap();
+        (*leaders).push(self.fsm.region_id());
```

Comment:
please do not collect it here. Make the force leader logic independent, you can collect it in `WaitApply`